### PR TITLE
`vault operator diagnose` stub command

### PR DESCRIPTION
**Pull request by mgritter**
This adds the new command to the CLI, and a flag to `vault server`.
It performs some light refactoring to allow us to see the storage setup status, but startup is not completely covered yet (nor meant to be.)

As part of Vault Server:
```
mgritter@ubuntu:~/vault$ VAULT_DIAGNOSE=1 bin/vault server -diagnose= -config=../vault-enterprise/diagnose-proto/test-config.hcl
Vault v1.6.0-dev ('cbbc0d815b131c97da59692a4cb0289f075d23e2+CHANGES')
[  ok  ] Parse configuration
[  ok  ] Access storage
Error parsing Seal configuration: Put "https://test-vault.example.com:8300/v1/transit/encrypt/autounseal": dial tcp 127.0.0.1:8300: connect: connection refused
2021-02-01T15:21:07.168-0800 [INFO]  proxy environment: http_proxy= https_proxy= no_proxy=
2021-02-01T15:21:10.951-0800 [INFO]  seal.transit: unable to renew token, disabling renewal: err="Put "https://test-vault.example.com:8300/v1/auth/token/renew-self": dial tcp 127.0.0.1:8300: connect: connection refused"
```

As standalone command:
```
mgritter@ubuntu:~/vault$ VAULT_DIAGNOSE=1 bin/vault operator diagnose -config bad-config.hcl
Vault v1.6.0-dev ('cbbc0d815b131c97da59692a4cb0289f075d23e2+CHANGES')
[  ok  ] Parse configuration
[failed] Access storage
Error initializing storage of type raft: failed to create fsm: failed to open bolt file: open /var/doesnotexist/raft/vault.db: no such file or directory
```

### CODE REVIEW

File: command/operator_diagnose.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/sdk/version"
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+)
+
+const OperatorDiagnoseEnableEnv = "VAULT_DIAGNOSE"
+
+var _ cli.Command = (*OperatorDiagnoseCommand)(nil)
```
Comment (swayne275) [inline command/operator_diagnose.go:14]: what does code of this form do again?

**Review by swayne275 — APPROVED** [review]

File: command/operator_diagnose.go (hunk 1 above)
Comment (mgritter) [reply command/operator_diagnose.go:14]: It causes a compile-time error if OperateDiagnoseCommand is not actually a cli.Command.  (In this exact circumstance it probably doesn't matter, but in other situations that might only be detected at runtime when a cast is attempted.)

Comment (swayne275) [reply command/operator_diagnose.go:14]: thanks! what does the `(nil)` syntax at the end of the line mean

Comment (mgritter) [reply command/operator_diagnose.go:14]: The other variation of this idiom.

```
var _ cli.Command = &OperatorDiagnoseCommand{}
```

which creates a real object.  But this version instead casts a nil pointer to the appropriate type.

Comment (swayne275) [reply command/operator_diagnose.go:14]: oh cool, thanks!

**Pull request merged**